students/rfelts/lesson04/mailroom.py

```diff
@@ -0,0 +1,204 @@
+    you email.
+    """
+
+    prompt_string = "\nPlease enter the donor's name: "
+    donor_name = input(prompt_string)
+
+    donor_list = create_name_list()
+
+    # Check for the donor's name (case insensitive) and updates or adds their donation history
+    if not any(s.lower() == donor_name.lower() for s in donor_list):
+        # Add the new donor and their donation
+        donor_data[donor_name] = [prompt_donation_amount(donor_name)]
+
+        # Pass in a dict with the required info for the message
+        print(compose_message({'name': donor_name, 'last_donation': donor_data.get(donor_name)[-1],
+                                       'total_donations': sum(donor_data.get(donor_name))}))
+    else:
+        for s in donor_list:
+            if s.lower() == donor_name.lower():
+                # Update the donor's donation history
+                donor_data[s].append(prompt_donation_amount(donor_name))
+
+                # Pass in a dict with the required info for the message
+                print(compose_message({'name': s, 'last_donation': donor_data.get(s)[-1],
+                                       'total_donations': sum(donor_data.get(s))}))
+                break
+
+
+def prompt_donation_amount(donor):
+    """
+    Prompt the user for the amount donated
+    :param donor - string representing the name of the person making the donation
+    :return: a float representing the amount donated
+    """
+
+    prompt_string = "\nPlease enter " + donor + "'s donation amount: "
+    response = input(prompt_string)
+
+    # Verify the value entered is numeric if not prompt the user for a new value
+    while True:
+        try:
+            response = float(response)
+            return response
+        except:
+            print("The value enter was invalid.")
+            response = input(prompt_string)
+
+
+def compose_message(donor_info):
+    """
+    Compose a thank you message listing the current/previous donation and the donor's donation total
+    :param donor_info - dict containing the donor's info: name, last donation, and total donation ammount
+    :return - a String containing the thank you message for the user
+    """
+
+    return("\nTo: {name}\nSubject: Thank you.\n\n{name} thank you for your previous generous donation of "
+           "{last_donation:<,.2f}.\nYou're total donations to date are now: {total_donations:<,.2f}.".format(**donor_info))
+
+
+def exit_menu():
+    """
+    Exits the current menu or program
+    :return - a String to exit
+    """
+
+    return "exit"
+
+
+# Main menu options to be presented to the user
+menu_options = ("\n1 - Send Thank You", "\n2 - Create Report", "\n3 - Send Letters to Everyone", "\n0 - Exit")
+menu_dict = {"0": exit_menu, "1": send_thank_you, "2": create_report, "3": send_to_everyone}
+
+
+if __name__ == '__main__':
+    prompt_user(menu_options, menu_dict)
+    print("\nProgram Over")
```

Comment:
Nice use of dictionaries in this.  Your function to write letters to files works well.  Nicely done!